Bearded.Monads/OptionExtensions.cs

```diff
@@ -310,16 +323,22 @@ public static A ElseNew<A>(this Option<A> items) where A : new()
             return await act(source.ForceValue());
         }
 
-        public static Option<IEnumerable<B>> Traverse<A, B>(this IEnumerable<A> enumerable, Func<A, Option<B>> callback)
+        [DebuggerStepThrough]
+        public static Option<IEnumerable<A>> AllOrNone<A>(this IEnumerable<Option<A>> options)
         {
-            var allOptions = enumerable.Select(callback).ToList();
+            var result = new List<A>();
+            return options.All(option => option.WhenSome(result.Add))
```

Comment:
I'm not sure how I feel about relying on the `true` operator here.

Also, this probably fits a bit more neatly with `Traverse`, e.g. `enumerable.Traverse(callback).Where(xs => xs.Count() == enumerable.Count())`, or, more optimally: `enumerable.Select(callback).All(x => x.IsSome).Traverse(id)`